src/blobtree/ScaleNode.js

```diff
@@ -0,0 +1,227 @@
+
+            this.aabb.expandByVector(new THREE.Vector3(x_scale,y_scale,z_scale));
+            this.valid_aabb = true;
+        }
+    };
+
+     /**
+     * @link Element.computeAABB for a complete description
+     */
+      computeAABB () {
+        this.aabb.makeEmpty();
+        for (var i = 0; i < this.children.length; i++) {
+            this.children[i].computeAABB();
+            this.aabb.union( this.children[i].getAABB());
+        }
+
+        let bb_size = new THREE.Vector3();
+        this.aabb.clone().getSize(bb_size);
+        let x_scale = bb_size.x*(this._scale.x - 1.0);
+        let y_scale = bb_size.y*(this._scale.y - 1.0);
+        let z_scale = bb_size.z*(this._scale.z - 1.0);
+
+        this.aabb.expandByVector(new THREE.Vector3(x_scale,y_scale,z_scale));
+    }
+
+    /**
+     *  @link Element.value for a complete description
+     *
+     *  @param {THREE.Vector3} p
+     *  @param {ValueResultType} res
+     */
+    value (p, res) {
+        // TODO : check that all bounding box of all children and subchildrens are valid
+        //        This enable not to do it in prim and limit the number of assert call (and string built)
+
+        var l = this.children.length;
+        var tmp = this.tmp_res;
+        tmp.g = res.g ? this.tmp_g : null;
+        tmp.m = res.m ? this.tmp_m : null;
+
+        // Init res
+        res.v = 0;
+        if (res.m) {
+            res.m.copy(Material.defaultMaterial);
+        } if (res.g) {
+            res.g.set(0, 0, 0);
+        } else if (res.step !== undefined) {
+            // that, is the max distance
+            // we want a value that loose any 'min'
+            res.step = 1000000000;
+        }
+
+
+        if (this.aabb.containsPoint(p) && l !== 0) {
+
+
+            let center = new THREE.Vector3();
+            this.aabb.getCenter(center);
+
+            let st_p =  new THREE.Vector3((p.x - center.x)/this._scale.x + center.x
+                                        ,(p.y - center.y)/ this._scale.y + center.y
+                                        ,(p.z - center.z)/ this._scale.z + center.z);
+
+            res.v = Number.MAX_VALUE;
+            for (var i = 0; i < l; ++i) {
+                this.children[i].value(st_p, tmp);
+                res.v = tmp.v;
+                if (res.g) {
+                    res.g.copy(tmp.g);
+                }
+                if (res.m) {
+                    res.m.copy(tmp.m);
+                }
+                // within primitive potential
+                if (res.step || res.stepOrtho) {
+                    throw "Not implemented";
+                }
+            }
+        }
+        else if (res.step || res.stepOrtho) {
+            throw "Not implemented";
+        }
+    }
+
+    /**
+     *  @link Element.trim for a complete description.
+     *
+     *  @param {THREE.Box3} aabb
+     *  @param {Array<Element>} trimmed
+     *  @param {Array<Node>} parents
+     */
+    trim(aabb, trimmed, parents) {
```

Comment:
I think this function is broken. Trimming is supposed to cut from the tree all children not influencing a given area.
So it should depend on the `_scale` parameter, shouldn't it ?